crt_portal/cts_forms/tests.py

```diff
@@ -1,28 +1,59 @@
-import datetime
-
 from django.test import TestCase
-from django.utils import timezone
-from django.urls import reverse
-
-from .models import ViolationReport
-
-
-def create_question(description_text, days):
-    """
-    Create a question with the given `description_text` and published the
-    given number of `days` offset to now (negative for questions published
-    in the past, positive for questions that have yet to be published).
-    """
-    time = timezone.now() + datetime.timedelta(days=days)
-    return ViolationReport.objects.create(description_text=description_text, create_date=time)
-
-class QuestionDetailViewTests(TestCase):
-    def test_past_question(self):
-        """
-        The detail view of a question with a pub_date in the past
-        displays the question's text.
-        """
-        past_question = create_question(description_text='Past Question.', days=-5)
-        url = reverse('crt_forms:detail', args=(past_question.id,))
-        response = self.client.get(url)
-        self.assertContains(response, past_question.description_text)
\ No newline at end of file
+
+from .models import ProtectedClass, Report
+from .forms import WhatHappened, Where, Who, Details, Contact
+
+class Valad_Form_Test(TestCase):
+    """Confirms each form is valid when given valid test data."""
```

Comment:
@LindsayYoung Let's add a test case with invalid data to make sure that the form isn't incorrectly marking all input data as valid. I think the current state of the form validation logic is that it's overly permissive — when looking at the form manually I found that the form didn't throw an error when I left required fields blank.

Reply:
Fixed! Thank you @LindsayYoung